Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                        {
+                            $frontDoorEp.DomainName = $domainName
+                            $frontDoorEp.FrontDoorName = $resourceName
+                            $frontDoorEp.MinTLSVersion = $MinTLSVersion
+                            $frontDoorEp.ResourceGroup = $resourceGroupName
+                            $frontDoorEp.TypeOfCertificate = $TypeOfCertificate
+                            $frontDoorEp.isMinTLSVersionSetOnCustomDomain = $false
+                        }
+                        else
+                        {
+                            $frontDoorsEndpointWithtReqMinTLSVersion = $item
+                        }
+                    }
+                    if($null -ne $frontDoorEp)
+                    {
+                       $frontDoorsEndpointWithoutReqMinTLSVersion =  $frontDoorEp | Select-Object @{N='ResourceGroupName';E={$frontDoorEp.ResourceGroup}},
+                                                                                    @{N='ResourceName';E={$frontDoorEp.FrontDoorName}},
+                                                                                    @{N='DomainName';E={$frontDoorEp.DomainName}},
+                                                                                    @{N='CertificateType';E={$frontDoorEp.TypeOfCertificate}},
+                                                                                    @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
+                                                                                    @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
+                    }
+                    
+                    else
+                    {
+                        Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+              }
+            }
+            catch
+            {
+                $frontDoorsSkipped += $frontDoorResource
+                Write-Host "Error fetching Front Door configuration: Resource Name: [$($resourceName)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
+                $logSkippedResources += $logResource
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    
+
+   foreach($item in $frontDoorsEndpointWithoutReqMinTLSVersion){
+        if (-not $frontDoorsEndpointWithoutReqMinTLSVersion)
```

Comment:
what's the benefit of this check inside foreach loop?

Reply:
Added condition there.